ttnn/cpp/ttnn/operations/data_movement/permute/device/permute_program_factory.cpp

```diff
@@ -14,8 +15,9 @@ uint32_t num_pages(const ttnn::Tensor& input_tensor) {
 }
 
 uint32_t page_size(const ttnn::Tensor& input_tensor) {
+    auto BUFFER_ALIGNMENT = input_tensor.buffer()->buffer_type() == tt::tt_metal::BufferType::DRAM ? DRAM_ALIGNMENT : L1_ALIGNMENT;
     const auto& padded_shape = input_tensor.get_logical_shape();  // in anticipation of RM padding
-    return padded_shape[-1] * input_tensor.element_size();
+    return tt::round_up(padded_shape[-1] * input_tensor.element_size(), BUFFER_ALIGNMENT);
```

Comment:
can you get page_size from `input_tensor.get_tensor_spec()`?
CC @sminakov-tt

Reply:
Ideally you should be able to ask the spec

Reply:
Page size is actually configurable based on the op.

My page sizes are actually a constant size here for my other kernel independent of inputs, which is what makes this kernel generic. We sacrifice some perf to ensure it scales for all inputs (I read in 64B chunks regardless of input size). A generic page size getter probably wouldn't work (though tbf, most kernels just set them equal to row width for rm.